Docs: Layout guidelines

hectoid opened the pull request:
### Summary
Added layout guidelines for mobile and tablet

#### What changed?

Added layout guidelines.
<img width="555" alt="image" src="https://github.com/pinterest/gestalt/assets/96082362/644bcc44-9c9a-4503-8ada-476194a5bf05">
<img width="1202" alt="image" src="https://github.com/pinterest/gestalt/assets/96082362/5385116d-e8fc-4124-b1de-7f58450e84b6">

Moved Form layout code to the Forms section.
<img width="1508" alt="image" src="https://github.com/pinterest/gestalt/assets/96082362/ffc451d4-70f0-44b6-a9c5-80fc6bc15a1b">


#### Why?

Designers ask for guidance on layout. We direct them to Lego guidelines in Dropbox. But Dropbox will soon be deprecated and many people already don't have access. So, we've moved them to Gestalt. The existing "layouts" page only had one layout for Forms, which makes more sense to include in the Forms section. We can revisit once we have sample code/templates for multiple layouts.

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-6422)
- [TDD](l[ink to Paper doc](https://paper.dropbox.com/doc/Lego-visual-language--CG348G4fLUFYO_J68DxZmm~NAg-16LVkb3NstMqJskFrXU22#:uid=803310587327265656450072&h2=Layout))
- [TDD update](https://docs.google.com/document/d/15oUsW2czdCtPC-G5X6RYpcx5R8pG7y36c6VaiGY_pPo/edit?usp=sharing)
- [Figma](https://www.figma.com/file/c7mskpSTbhp9NneVSdnJBj/Doc---Layout?type=design&node-id=1-9049&mode=design&t=EyGGMH05jRtREYYd-11)

### Checklist

- [x] Added unit and Flow Tests
- [x] Added documentation + accessibility tests
- [x] Verified accessibility: keyboard & screen reader interaction
- [x] Checked dark mode, responsiveness, and right-to-left support
- [ ] Checked stakeholder feedback (e.g. Gestalt designers)

diyorbek commented:
LGTM

rlingineni commented on .vscode/settings.json:
```diff
@@ -1,8 +1,8 @@
   "editor.codeActionsOnSave": {
-    "source.fixAll.eslint": true,
-    "source.fixAll.shellcheck": true
+    "source.fixAll.eslint": "explicit",
```
@hectoid  - these vsocde changes seem to be back

rlingineni commented:
@hectoid  - can you also add a [redirect](https://github.com/pinterest/gestalt/blob/319fed128eecc4ddb0fba68ad99f12c3bdd679ef/docs/redirects.js#L114) in this file since we're removing the other link

hectoid commented:
> @hectoid - can you also add a [redirect](https://github.com/pinterest/gestalt/blob/319fed128eecc4ddb0fba68ad99f12c3bdd679ef/docs/redirects.js#L114) in this file since we're removing the other link

good catch @rlingineni ! added. also fixed the settings.js issues. I got a lot of errors locally this time around, so hopefully it's just an issue with vs code again. I'm not sure why that settings file keeps changing.
